ui/src/components/setlist/SetlistEditorContainer.tsx

```diff
@@ -194,13 +206,33 @@ const SetlistEditorContainer: FC<SetlistEditorProps> = () => {
       //   }
 
       // })
-
       if (payload.status === 200) {
+        // Updates the ownership and folder
+        Promise.all(
+          setlistFolderIds.map((folderId) => {
+            const currentFolder = folderOptions.find((folder) => folder._id === folderId);
+            // Only updates the folder the user have access to
+            if (currentFolder) {
+              axios.put('/api/groups/update', {
+                id: folderId,
+                setlistIds: [...(currentFolder?.setlistIds || []), payload.data._id],
+              });
+            }
+          })
+        );
+
+        if (!ownership.setlistIds.some((setlist) => setlist.id === payload.data._id)) {
```

Comment:
i think ideally the operation to update ownership shouldnt be exposed like this. 
may be a better way of doing it is calling the /api/.groups/update api will trigger a server side operation to change the ownership